### PR TITLE
[intel-npu] Adding private properties tests

### PR DESCRIPTION
**Pull request by alexandruenache1111**
### Details:
Adding private properties tests for npu.

Also, skipping some tests if `ELF backend` is set to _NO_ on Linux. Graphfile does not work on Linux, so there is no substitute for `ELF backend`.

### Tickets:
 - *E-118136*

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
Please, resolve issues with excessive commits in this PR.

**Comment by alexandruenache1111** [comment]
> Please, resolve issues with excessive commits in this PR.

Resolving now

File: src/plugins/intel_npu/tests/functional/internal/overload/ov_plugin/internal_properties_tests.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    void SetUp() override;
+
+    void TearDown() override;
+
+    AnyMap compileModelProperties;
+};
+
+using OVCheckSetSupportedRWMetricsPropsTestsNPU = OVPropertiesTestsWithCompileModelPropsNPU;
+
+}  // namespace ov::test::behavior
```
Comment (pereanub) [inline src/plugins/intel_npu/tests/functional/internal/overload/ov_plugin/internal_properties_tests.hpp:47]: new line

File: src/plugins/intel_npu/tests/functional/shared_tests_instances/behavior/ov_plugin/internal_properties_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                                            ::testing::ValuesIn(IncorrectMutablePropertiesWrongValueTypes)),
+                         (ov::test::utils::appendPlatformTypeTestName<OVPropertiesIncorrectTestsNPU>));
+
+INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests,
+                         OVCheckSetSupportedRWMetricsPropsTestsNPU,
+                         ::testing::Combine(::testing::Values(ov::test::utils::DEVICE_NPU),
+                                            ::testing::ValuesIn(CorrectPluginMutableProperties)),
+                         (ov::test::utils::appendPlatformTypeTestName<OVCheckSetSupportedRWMetricsPropsTestsNPU>));
+
+}  // namespace ov::test::behavior
```
Comment (pereanub) [inline src/plugins/intel_npu/tests/functional/shared_tests_instances/behavior/ov_plugin/internal_properties_tests.cpp:184]: new line

**Review by MirceaDan99 — COMMENTED** [review]
If this is an internal tests, I think we should avoid placing it in `shared_tests_instances`. A better place I think it will be `src/plugins/intel_npu/tests/functional/behavior/ov_plugin/internal_properties_tests.hpp`

File: src/plugins/intel_npu/tests/functional/shared_tests_instances/behavior/ov_plugin/internal_properties_tests.cpp (hunk 2 above)
Comment (alexandruenache1111) [reply src/plugins/intel_npu/tests/functional/shared_tests_instances/behavior/ov_plugin/internal_properties_tests.cpp]: Thanks for pointing out. I added the modifications

**Review by MirceaDan99 — APPROVED** [review]

**Review by sbutnari — APPROVED** [review]

**Pull request merged**